wgsl/index.bs

```diff
@@ -4016,13 +4016,17 @@ then the last statement in the function body must be a [=return=] statement.
 function_decl
   : attribute_list* function_header compound_statement
 
-function_type_decl
+function_header
+  : FN IDENT PAREN_LEFT param_list PAREN_RIGHT function_return_type_decl_optional
+
+function_return_type_decl_optional
```

Comment:
The spec style seems to prefer (or at least has a non-trivial number of instances) having:
```
function_header
  : FN IDENT PAREN_LEFT param_list PAREN_RIGHT (ARROW function_return_type_decl)?
```

Editorially we should decide which way to go, but I don't think it needs to hold up this PR.